frontend/src/components/QueryBuilderHandler/QueryBuilderHandler.tsx

```diff
@@ -140,6 +143,17 @@ class QueryBuilderHandler extends React.Component<QueryBuilderHandlerProps> {
         />
       );
     }
+    if (query === 'advanced') {
+      const columns = source.get('columns') as ColumnList;
+      const selectableColumns = getStepSelectableColumns(step, steps, columns) as Set<string>;
+
+      const onTextareaChange = (options: string) => {
+        const queryObject = parseAdvancedQueryString(options, selectableColumns);
+        onUpdateOptions(JSON.stringify(queryObject));
+      };
+
+      return <BasicTextarea onChange={onTextareaChange} alerts={alerts} />;
```

Comment:
Each of the options in this function was neat ... e.g. `SelectQueryBuilder`, `FilterQueryBuilder`, `TransformQueryBuilder` ... then we have this ... why is you crowding our nice code? Basically, create a component to handle all your advanced filter stuff ... I expect it to feed into the `onUpdateOptions` function since it too will return JSON. In this component, I don't want to know how an advanced filter is built.

Reply:
Created `TextFilterQueryBuilder` to build text filters